src/Symfony/Component/Intl/Tests/TimezonesTest.php

```diff
@@ -562,4 +573,67 @@ public function testForCountryCode()
         $this->assertSame(['Europe/Amsterdam'], Timezones::forCountryCode('NL'));
         $this->assertSame(['Europe/Berlin', 'Europe/Busingen'], Timezones::forCountryCode('DE'));
     }
+
+    /**
+     * @expectedException \Symfony\Component\Intl\Exception\MissingResourceException
+     */
+    public function testForCountryCodeWithUnknownCountry()
+    {
+        Timezones::forCountryCode('foobar');
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Intl\Exception\MissingResourceException
+     */
+    public function testGetCountryCodeWithUnknownTimezone()
+    {
+        Timezones::getCountryCode('foobar');
+    }
+
+    /**
+     * @dataProvider provideTimezones
+     */
+    public function testGetGmtOffsetAvailability(string $timezone)
+    {
+        $this->assertInternalType('int', Timezones::getRawOffset($timezone));
+        $this->assertInternalType('string', Timezones::getGmtOffset($timezone));
+    }
+
+    /**
+     * @dataProvider provideTimezones
+     */
+    public function testGetCountryCodeAvailability(string $timezone)
+    {
+        try {
+            $this->assertInternalType('string', Timezones::getCountryCode($timezone));
+        } catch (MissingResourceException $e) {
```

Comment:
if it is possible for valid timezone ids to throw a MissingResourceException when trying to get their code, we need to document that in the interface, as it becomes necessary to catch it when using it (or maybe the Timezones class should handle that internally)

Reply:
see https://github.com/symfony/symfony/pull/31332